### ISSUE TITLE
SudoDatabaseTransactions

### ISSUE THREAD

**Issue by devinfd** [laravel/dusk#148]
_tldr: no issue just sharing._

I'm posting this here to share my `SudoDatabaseTransactions` trait. I created this a few years ago because I had tests against mysql tables with full text search columns which require non-transactional commits. I didn't want to fill the table full of garbage so at the beginning of each test the `SudoDatabaseTransactions` trait just truncates them.

It's understandable that the DatabaseTransactions trait would not work in Dusk and my app has lots of migrations so I wasn't willing to use the DatabaseMigrations trait and wait around before each test ran. So I was happy to discover that this trait works perfectly with Dusk as well. I think it might be too risky to merge this into Dusk because of the damage it can cause if the user wasn't paying attention. But some folks might find it handy so here it is.

### WARNING
**The `SudoDatabaseTransactions` trait is dangerous because it truncates your data. Use it wisely.**

```php
namespace Tests;

use Illuminate\Support\Facades\DB;

trait SudoDatabaseTransactions
{
    /**
     * Tables to exclude from truncation.
     *
     * @var array
     */
    protected $excludeableTables = [
        'migrations'
    ];

    /**
     * @before
     */
    public function sudoDatabaseTransaction()
    {
        static $tables = [];

        if (empty($tables)) {
            $tables = $this->getTables();
        }

        $this->beforeApplicationDestroyed(function () use ($tables) {
            DB::statement('SET FOREIGN_KEY_CHECKS=0');

            foreach ($tables as $table) {
                DB::table($table)->truncate();
            }

            DB::statement('SET FOREIGN_KEY_CHECKS=1');
        });
    }

    /**
     * Get the databases tables excluding the excludeableTables.
     *
     * @return \Illuminate\Support\Collection
     */
    private function getTables()
    {
        return DB::table('information_schema.tables')
                ->where('table_schema', app('db')->getDatabaseName())
                ->whereNotIn('table_name', $this->excludeableTables)
                ->pluck('table_name');
    }
}
```

```php
namespace Tests\Browser;

use Tests\DuskTestCase;
use Tests\SudoDatabaseTransactions;

class AdmitPatientTest extends DuskTestCase
{
    use SudoDatabaseTransactions;

    /** @test */
    public function example_test()
    {
        # code...
    }
}
```

**Comment by deleugpn** [comment]
Much like `php artisan db:nuke`, the damage potential can be mitigated by reassuring the user of what he's doing. IMHO, the problem with merging this into Dusk is not about data loss. It's just that it's completely coupled with MySQL only while Eloquent supports more databases.
